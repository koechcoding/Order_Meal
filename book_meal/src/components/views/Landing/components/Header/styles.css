
header.welcome {
    min-height: 80vh;
    background: url("./assets/bg.jpg") no-repeat center bottom;
    background-color: #333;
    background-size: cover;
    text-align: center;
    width: 100%; 
}
header.welcome .logo {
    margin-top: 10vh; 
}
header.welcome #slogan {
    padding: 12px;
    font-size: 30px;
    background-color: rgba(0, 80, 80, 0.7); }
    header.welcome #slogan span span {
    color: #f5ee9e; 
}