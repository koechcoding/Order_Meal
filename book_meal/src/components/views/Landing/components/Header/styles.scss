header.welcome {
    min-height: 80vh;
    background: url('./assets/bg.jpg') no-repeat center bottom;
    background-color: #333;
    background-size: cover;
    text-align: center;
    width: 100%;
}
.logo {
    margin-top: 10vh;
}
#slogan {
    padding: 12px;
    font-size: 30px;
    background-color: rgba(0, 80, 80, 0.7);
}

span span {
    color: #f5ee9e;
}