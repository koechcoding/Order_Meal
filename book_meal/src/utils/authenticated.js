export default function authenticated(){
    try{
        return JSON.parse(localStorage.getItem('user'));
    } catch(err){
        
    }
}