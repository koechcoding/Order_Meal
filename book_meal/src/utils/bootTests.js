import Enzyme from "enzyme";
import Adapter from "enzyme-adapter-react-16";
Enzyme.configure({ adapter: new Adapter()});

global.window = {}
import localStorage from 'mock-local-Storage';
window.localStorage = global.localStorage

export default Enzyme;
