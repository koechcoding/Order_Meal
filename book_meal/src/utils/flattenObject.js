export default function flattenObject(obj){
    let result = {}
    for (const key in obj){
        if((typeof obj[key])=== 'object' &&  obj[key]  !== null) {
            let flattened = flattenObject(obj[key])
            for(const key2 in flattened){
                result[key + "." + key2] = flattened[key2];
            }
        }else{
            result[key] = obj[key];
        }
    }
    return result;
}