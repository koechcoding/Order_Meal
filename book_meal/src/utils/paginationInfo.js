export default function paginationInfo(page){

    if(!page){
        return{}
    }

    let currentPage = 1
    if(page.next_page)
         currentPage = page.next_page -1;
    else if (page.prev_page)
        currentPage = page.prev_page + 1;

    let pageCount = Math.ceil(page.total / page.per_paage);
    if(Number.isNaN(pageCount) || pageCount === 0)
      pageCount = 1; 

    return {
        pageCount,
        currentPage,
        total: page.total,
        perPage: page.per_page,
        hasNext: page.has_next,
        hasPrev: page.has_prev,
        nextPage: page.next_page,
        prevPage: page.prev_page,
        currentCount: page.current_count,
    };
}