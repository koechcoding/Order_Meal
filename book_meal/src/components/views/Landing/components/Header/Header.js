import React from 'react';
import { Link} from 'react-router-dom';
import Typed from 'react-typed';
import './styles.css';

class Header extends React.Component{
     render(){
        const strings =[
            "Welcome to yummy stuff!",
            "Welcome to where food lives!",
            "Welcome to book a meal."
        ];
        
        const settings = {
            strings,
            startDelay: 200,
            typeSpeed: 50,
            backSpeed: 40,
            showCursor: false,
        }

        return(
            <header className="welcome">
                <nav className='container'>
                    <div className="row pt-md-5 pt-0">
                    <div className="col-12 col-md-8 offset-md-2 pt-0">
                            <h1 className="animated fadeInUp logo text-flav pb-md-3">
                                BAM!
                            </h1>
                            <h3 id="slogan" className="animated fadeInDown text-flav mb-md-4">
                                <Typed {...settings} />
                            </h3>
                        </div>
                    </div>
                    <div className='row mt-md-2 mt-2'>
                    <div className="col-6">
                            <Link to="/sign-up" className="animated fadeInLeft btn btn-primary btn-lg float-right">
                                Sign Up
                            </Link>
                        </div>
                        <div className="col-6">
                            <Link to="/login" className="animated fadeInRight btn btn-primary btn-lg float-left">
                                Log In
                            </Link>
                        </div>
                    </div>
                </nav>
            </header>
        );
     }
}

export default Header;