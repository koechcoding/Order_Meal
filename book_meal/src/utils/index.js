import singleError from './singleError';
import paginationInfo from './paginationInfo';
import authenticated from './authenticated';
import flattenObject from './flattenObject';

export { 
    singleError, 
    paginationInfo, 
    authenticated,
    flattenObject
}