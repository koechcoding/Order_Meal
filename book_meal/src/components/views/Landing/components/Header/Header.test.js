import React from 'react';
import { shallow }  from 'enzyme';
import 'src/utils/bootTests'; 
import Header from './Header';

describe('<Header />', () => {
    it('renders correctly', () => {
        const wrapper = shallow(<Header/>);
        expect(wrapper).toMatchSnapshot();
    });
});