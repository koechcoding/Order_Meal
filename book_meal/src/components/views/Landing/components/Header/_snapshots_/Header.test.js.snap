// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <nav
        className="container"
>
        <div
                className="row pt-md-5 pt-0"
        >
                <div
                        className="col-12 col-md-8 offset-md-2 pt-0"
                >
                        <h1
                                className="animated fadeInUp logo text-flav pb-md-3"
                        >
                                BAM!
                        </h1>
                        <h3
                                className="animated fadeInDown text-flav mb-md-4"
                                id="slogan"
                        >
                                <ReactTyped
                                        backSpeed={40}
                                        showCursor={false}
                                        startDelay={200}
                                        strings={
                                                Array [
                                                        "Welcome to yummy stuff!",
                                                        "Welcome to where food lives!",
                                                        "Welcome to book a meal.",
                                                      ]
                                        }
                                        typeSpeed={50}
                                />
                        </h3>
                </div>
        </div>
        <div
                className="row mt-md-2 mt-2"
        >
                <div
                        className="col-6"
                >
                        <Link
                                className="animated fadeInLeft btn btn-primary btn-lg float-right"
                                replace={false}
                                to="/sign-up"
                        >
                                Sign Up
                        </Link>
                </div>
                <div
                        className="col-6"
                >
                        <Link
                                className="animated fadeInRight btn btn-primary btn-lg float-left"
                                replace={false}
                                to="/login"
                        >
                                Log In
                        </Link>
                </div>
        </div>
</nav>,
      "className": "welcome",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="row pt-md-5 pt-0"
>
            <div
                        className="col-12 col-md-8 offset-md-2 pt-0"
            >
                        <h1
                                    className="animated fadeInUp logo text-flav pb-md-3"
                        >
                                    BAM!
                        </h1>
                        <h3
                                    className="animated fadeInDown text-flav mb-md-4"
                                    id="slogan"
                        >
                                    <ReactTyped
                                                backSpeed={40}
                                                showCursor={false}
                                                startDelay={200}
                                                strings={
                                                            Array [
                                                                        "Welcome to yummy stuff!",
                                                                        "Welcome to where food lives!",
                                                                        "Welcome to book a meal.",
                                                                      ]
                                                }
                                                typeSpeed={50}
                                    />
                        </h3>
            </div>
</div>,
          <div
            className="row mt-md-2 mt-2"
>
            <div
                        className="col-6"
            >
                        <Link
                                    className="animated fadeInLeft btn btn-primary btn-lg float-right"
                                    replace={false}
                                    to="/sign-up"
                        >
                                    Sign Up
                        </Link>
            </div>
            <div
                        className="col-6"
            >
                        <Link
                                    className="animated fadeInRight btn btn-primary btn-lg float-left"
                                    replace={false}
                                    to="/login"
                        >
                                    Log In
                        </Link>
            </div>
</div>,
        ],
        "className": "container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="col-12 col-md-8 offset-md-2 pt-0"
>
              <h1
                            className="animated fadeInUp logo text-flav pb-md-3"
              >
                            BAM!
              </h1>
              <h3
                            className="animated fadeInDown text-flav mb-md-4"
                            id="slogan"
              >
                            <ReactTyped
                                          backSpeed={40}
                                          showCursor={false}
                                          startDelay={200}
                                          strings={
                                                        Array [
                                                                      "Welcome to yummy stuff!",
                                                                      "Welcome to where food lives!",
                                                                      "Welcome to book a meal.",
                                                                    ]
                                          }
                                          typeSpeed={50}
                            />
              </h3>
</div>,
            "className": "row pt-md-5 pt-0",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h1
                  className="animated fadeInUp logo text-flav pb-md-3"
>
                  BAM!
</h1>,
                <h3
                  className="animated fadeInDown text-flav mb-md-4"
                  id="slogan"
>
                  <ReactTyped
                                    backSpeed={40}
                                    showCursor={false}
                                    startDelay={200}
                                    strings={
                                                      Array [
                                                                        "Welcome to yummy stuff!",
                                                                        "Welcome to where food lives!",
                                                                        "Welcome to book a meal.",
                                                                      ]
                                    }
                                    typeSpeed={50}
                  />
</h3>,
              ],
              "className": "col-12 col-md-8 offset-md-2 pt-0",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "BAM!",
                  "className": "animated fadeInUp logo text-flav pb-md-3",
                },
                "ref": null,
                "rendered": "BAM!",
                "type": "h1",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <ReactTyped
                    backSpeed={40}
                    showCursor={false}
                    startDelay={200}
                    strings={
                                        Array [
                                                            "Welcome to yummy stuff!",
                                                            "Welcome to where food lives!",
                                                            "Welcome to book a meal.",
                                                          ]
                    }
                    typeSpeed={50}
/>,
                  "className": "animated fadeInDown text-flav mb-md-4",
                  "id": "slogan",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "backSpeed": 40,
                    "showCursor": false,
                    "startDelay": 200,
                    "strings": Array [
                      "Welcome to yummy stuff!",
                      "Welcome to where food lives!",
                      "Welcome to book a meal.",
                    ],
                    "typeSpeed": 50,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": "h3",
              },
            ],
            "type": "div",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="col-6"
>
                <Link
                                className="animated fadeInLeft btn btn-primary btn-lg float-right"
                                replace={false}
                                to="/sign-up"
                >
                                Sign Up
                </Link>
</div>,
              <div
                className="col-6"
>
                <Link
                                className="animated fadeInRight btn btn-primary btn-lg float-left"
                                replace={false}
                                to="/login"
                >
                                Log In
                </Link>
</div>,
            ],
            "className": "row mt-md-2 mt-2",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <Link
                  className="animated fadeInLeft btn btn-primary btn-lg float-right"
                  replace={false}
                  to="/sign-up"
>
                  Sign Up
</Link>,
                "className": "col-6",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Sign Up",
                  "className": "animated fadeInLeft btn btn-primary btn-lg float-right",
                  "replace": false,
                  "to": "/sign-up",
                },
                "ref": null,
                "rendered": "Sign Up",
                "type": [Function],
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <Link
                  className="animated fadeInRight btn btn-primary btn-lg float-left"
                  replace={false}
                  to="/login"
>
                  Log In
</Link>,
                "className": "col-6",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Log In",
                  "className": "animated fadeInRight btn btn-primary btn-lg float-left",
                  "replace": false,
                  "to": "/login",
                },
                "ref": null,
                "rendered": "Log In",
                "type": [Function],
              },
              "type": "div",
            },
          ],
          "type": "div",
        },
      ],
      "type": "nav",
    },
    "type": "header",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <nav
          className="container"
>
          <div
                    className="row pt-md-5 pt-0"
          >
                    <div
                              className="col-12 col-md-8 offset-md-2 pt-0"
                    >
                              <h1
                                        className="animated fadeInUp logo text-flav pb-md-3"
                              >
                                        BAM!
                              </h1>
                              <h3
                                        className="animated fadeInDown text-flav mb-md-4"
                                        id="slogan"
                              >
                                        <ReactTyped
                                                  backSpeed={40}
                                                  showCursor={false}
                                                  startDelay={200}
                                                  strings={
                                                            Array [
                                                                      "Welcome to yummy stuff!",
                                                                      "Welcome to where food lives!",
                                                                      "Welcome to book a meal.",
                                                                    ]
                                                  }
                                                  typeSpeed={50}
                                        />
                              </h3>
                    </div>
          </div>
          <div
                    className="row mt-md-2 mt-2"
          >
                    <div
                              className="col-6"
                    >
                              <Link
                                        className="animated fadeInLeft btn btn-primary btn-lg float-right"
                                        replace={false}
                                        to="/sign-up"
                              >
                                        Sign Up
                              </Link>
                    </div>
                    <div
                              className="col-6"
                    >
                              <Link
                                        className="animated fadeInRight btn btn-primary btn-lg float-left"
                                        replace={false}
                                        to="/login"
                              >
                                        Log In
                              </Link>
                    </div>
          </div>
</nav>,
        "className": "welcome",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="row pt-md-5 pt-0"
>
              <div
                            className="col-12 col-md-8 offset-md-2 pt-0"
              >
                            <h1
                                          className="animated fadeInUp logo text-flav pb-md-3"
                            >
                                          BAM!
                            </h1>
                            <h3
                                          className="animated fadeInDown text-flav mb-md-4"
                                          id="slogan"
                            >
                                          <ReactTyped
                                                        backSpeed={40}
                                                        showCursor={false}
                                                        startDelay={200}
                                                        strings={
                                                                      Array [
                                                                                    "Welcome to yummy stuff!",
                                                                                    "Welcome to where food lives!",
                                                                                    "Welcome to book a meal.",
                                                                                  ]
                                                        }
                                                        typeSpeed={50}
                                          />
                            </h3>
              </div>
</div>,
            <div
              className="row mt-md-2 mt-2"
>
              <div
                            className="col-6"
              >
                            <Link
                                          className="animated fadeInLeft btn btn-primary btn-lg float-right"
                                          replace={false}
                                          to="/sign-up"
                            >
                                          Sign Up
                            </Link>
              </div>
              <div
                            className="col-6"
              >
                            <Link
                                          className="animated fadeInRight btn btn-primary btn-lg float-left"
                                          replace={false}
                                          to="/login"
                            >
                                          Log In
                            </Link>
              </div>
</div>,
          ],
          "className": "container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <div
                className="col-12 col-md-8 offset-md-2 pt-0"
>
                <h1
                                className="animated fadeInUp logo text-flav pb-md-3"
                >
                                BAM!
                </h1>
                <h3
                                className="animated fadeInDown text-flav mb-md-4"
                                id="slogan"
                >
                                <ReactTyped
                                                backSpeed={40}
                                                showCursor={false}
                                                startDelay={200}
                                                strings={
                                                                Array [
                                                                                "Welcome to yummy stuff!",
                                                                                "Welcome to where food lives!",
                                                                                "Welcome to book a meal.",
                                                                              ]
                                                }
                                                typeSpeed={50}
                                />
                </h3>
</div>,
              "className": "row pt-md-5 pt-0",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <h1
                    className="animated fadeInUp logo text-flav pb-md-3"
>
                    BAM!
</h1>,
                  <h3
                    className="animated fadeInDown text-flav mb-md-4"
                    id="slogan"
>
                    <ReactTyped
                                        backSpeed={40}
                                        showCursor={false}
                                        startDelay={200}
                                        strings={
                                                            Array [
                                                                                "Welcome to yummy stuff!",
                                                                                "Welcome to where food lives!",
                                                                                "Welcome to book a meal.",
                                                                              ]
                                        }
                                        typeSpeed={50}
                    />
</h3>,
                ],
                "className": "col-12 col-md-8 offset-md-2 pt-0",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "BAM!",
                    "className": "animated fadeInUp logo text-flav pb-md-3",
                  },
                  "ref": null,
                  "rendered": "BAM!",
                  "type": "h1",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <ReactTyped
                      backSpeed={40}
                      showCursor={false}
                      startDelay={200}
                      strings={
                                            Array [
                                                                  "Welcome to yummy stuff!",
                                                                  "Welcome to where food lives!",
                                                                  "Welcome to book a meal.",
                                                                ]
                      }
                      typeSpeed={50}
/>,
                    "className": "animated fadeInDown text-flav mb-md-4",
                    "id": "slogan",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "backSpeed": 40,
                      "showCursor": false,
                      "startDelay": 200,
                      "strings": Array [
                        "Welcome to yummy stuff!",
                        "Welcome to where food lives!",
                        "Welcome to book a meal.",
                      ],
                      "typeSpeed": 50,
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  "type": "h3",
                },
              ],
              "type": "div",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div
                  className="col-6"
>
                  <Link
                                    className="animated fadeInLeft btn btn-primary btn-lg float-right"
                                    replace={false}
                                    to="/sign-up"
                  >
                                    Sign Up
                  </Link>
</div>,
                <div
                  className="col-6"
>
                  <Link
                                    className="animated fadeInRight btn btn-primary btn-lg float-left"
                                    replace={false}
                                    to="/login"
                  >
                                    Log In
                  </Link>
</div>,
              ],
              "className": "row mt-md-2 mt-2",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <Link
                    className="animated fadeInLeft btn btn-primary btn-lg float-right"
                    replace={false}
                    to="/sign-up"
>
                    Sign Up
</Link>,
                  "className": "col-6",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "Sign Up",
                    "className": "animated fadeInLeft btn btn-primary btn-lg float-right",
                    "replace": false,
                    "to": "/sign-up",
                  },
                  "ref": null,
                  "rendered": "Sign Up",
                  "type": [Function],
                },
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <Link
                    className="animated fadeInRight btn btn-primary btn-lg float-left"
                    replace={false}
                    to="/login"
>
                    Log In
</Link>,
                  "className": "col-6",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "Log In",
                    "className": "animated fadeInRight btn btn-primary btn-lg float-left",
                    "replace": false,
                    "to": "/login",
                  },
                  "ref": null,
                  "rendered": "Log In",
                  "type": [Function],
                },
                "type": "div",
              },
            ],
            "type": "div",
          },
        ],
        "type": "nav",
      },
      "type": "header",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;